Revit_Core_Engine/Convert/Environment/FromRevit/CableTray.cs

```diff
@@ -94,6 +152,20 @@ public static List<BH.oM.MEP.System.CableTray> CableTrayFromRevit(this Autodesk.
             return bhomCableTrays;
         }
 
+        /***************************************************/
+        /****              Private Methods              ****/
+        /***************************************************/
+        
+        [Description("Re-orders converted BHoM Line list to match the direction from the reference MEPCurve.")]
+        private static List<BH.oM.Geometry.Line> MatchRevitOrder(List<BH.oM.Geometry.Line> linesToMatch, MEPCurve reference)
+        {
+            LocationCurve locationCurve = reference.Location as LocationCurve;
+            Curve curve = locationCurve.Curve;
+            BH.oM.Geometry.Point referenceStart = curve.GetEndPoint(0).PointFromRevit();
+
+            return linesToMatch.OrderBy(x => x.Start.Distance(referenceStart)).ToList();
```

Comment:
Is that a robust solution? What if the curves had some weird snail-like shape? Having said that, couldn't you joint the curves into a `PolyCurve`, flip it so that its start point is closer to the start point of element's location curve and then explode?

A bit of an overhead, but may be worth - please dismiss this comment if that does not make sense to you.